### PR TITLE
[BugFix] Fix tablet ptr leak in compaction manager after it dropped

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

- [ ] I have checked the version labels which the pr will be auto backported to target branch

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/storage/tablet.h [hunk 1]
```diff
@@ -313,6 +315,7 @@ class Tablet : public BaseTablet {
     // compaction related
     std::unique_ptr<CompactionContext> _compaction_context;
     std::shared_ptr<CompactionTask> _compaction_task;
+    bool _disable_compaction = false;
```
Comment (chaoyli) [inline be/src/storage/tablet.h:318]:
```suggestion
    bool _enable_compaction = true;
```
Better to use positive sentence.

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by meegoo** [comment]
@mergify backport branch-2.5

**Comment by meegoo** [comment]
@mergify backport branch-2.5

**Pull request merged**